orderedsets/__init__.py

```diff
@@ -121,59 +123,58 @@ def __len__(self) -> int:
     def __contains__(self, o: object) -> bool:
         return o in self._dict
 
-    def __iter__(self) -> Iterator:  # type: ignore[type-arg]
+    def __iter__(self) -> Iterator[T]:
         return iter(self._dict)
 
-    def __and__(self, s: Set) -> OrderedSet:  # type: ignore[type-arg]
+    def __and__(self, s: Set[T]) -> OrderedSet[T]:
         return self.intersection(s)
 
-    def __iand__(self, s: Set) -> OrderedSet:  # type: ignore[type-arg]
+    def __iand__(self, s: Set[T]) -> OrderedSet[T]:
         result = self.intersection(s)
         self._dict = result._dict
         return result
 
-    def __or__(self, s: Set) -> OrderedSet:  # type: ignore[type-arg]
+    def __or__(self, s: Set[Any]) -> OrderedSet[T]:
         return self.union(s)
 
-    def __ior__(self, s: Set) -> OrderedSet:  # type: ignore[type-arg]
+    def __ior__(self, s: Set[Any]) -> OrderedSet[T]:
         result = self.union(s)
         self._dict = result._dict
         return result
```

Comment:
What are those `Any`s doing here?

Reply:
That's the only way I could get mypy to accept the typing on the `or`-related functions. Their signature in https://github.com/python/mypy/blob/5506cba158d76cd11697d1178d73a552aa617b7c/mypy/typeshed/stdlib/typing.pyi#L560 looks a bit different from the other functions, which might be the reason for the difference.